src/data/polynomial/erase_lead.lean

```diff
@@ -96,6 +96,27 @@ begin
   exact card_lt_card (erase_ssubset $ nat_degree_mem_support_of_nonzero h)
 end
 
+lemma erase_lead_card_support {R : Type*} [semiring R] {c : ℕ} {f : polynomial R}
+  (fc : f.support.card = c + 1) :
+  f.erase_lead.support.card = c :=
+begin
+  by_cases f0 : f = 0,
+  { rw [f0, support_zero, card_empty] at fc,
+    exact (c.succ_ne_zero fc.symm).elim },
+  { rw [erase_lead_support, card_erase_of_mem (nat_degree_mem_support_of_nonzero f0), fc],
+    exact c.pred_succ },
+end
+
+lemma erase_lead_card_support' {R : Type*} [semiring R] {c : ℕ} {f : polynomial R}
```

Comment:
The proof of this lemma doesn't even use `erase_lead_card_support` and it's shorter!
So you should put this one first, and use it to prove the other one, which should then be a one liner.
I'll leave it up to you to decide which one should get a `'` in its name.

Reply:
Point taken!  I only left the lemma that you suggested, since a proof of the other one was simply an application of the previous one!

So, the lemma that had a `'` is now the only one, with the `'` removed.  The other lemma disappeared.

Reply:
I haven't tried it out in Lean so I could be wrong, but the other variant isn't exactly an application of this one, right? Sometimes it can be useful to state the same (or almost the same) lemma with slightly different hypotheses to avoid fiddly rewrites.

Reply:
I just tried this: the code below compiles in Lean without errors.

```
lemma erase_lead_card_support' {R : Type*} [semiring R] {c : ℕ} {f : polynomial R}
  (fc : f.support.card = c + 1) :
  f.erase_lead.support.card = c :=
erase_lead_card_support fc
```

Reply:
Cool, that makes sense!

I just did some experimentation with `rw` and it seems that the primed variant may be worth keeping around for that, since rewriting with `erase_lead_card_support fc` where `(fc : f.support.card = c + 1)` ends up with `c + 1 - 1` which can be annoying. Also rewriting in reverse with `← erase_lead_card_support fc` doesn't work well for the same reason (as it wants `c + 1 - 1` instead of `c`).

So I suggest adding back `erase_lead_card_support'` after all. Feel free to merge after doing so.

bors d+